packages/cli/src/aws/update-bucket-policies.ts

```diff
@@ -0,0 +1,158 @@
+
+export interface UpdateBucketPoliciesOptions {
+  dryrun?: boolean;
+}
+
+interface Policy {
+  Version?: string;
+  Statement?: PolicyStatement[];
+}
+
+interface PolicyStatement {
+  Effect?: string;
+  Principal?: { AWS: string };
+  Action?: string | string[];
+  Resource?: string | string[];
+}
+
+/**
+ * The AWS "update-bucket-policies" command adds necessary policy statements to S3 bucket policy documents.
+ *
+ * This is necessary for Medplum deployments outside of the us-east-1 region.
+ *
+ * @param tag The Medplum stack tag.
+ * @param options The update options.
+ */
+export async function updateBucketPoliciesCommand(tag: string, options: UpdateBucketPoliciesOptions): Promise<void> {
+  const config = readConfig(tag);
+  if (!config) {
+    console.log('Config not found');
+    return;
+  }
+
+  const details = await getStackByTag(tag);
+  if (!details) {
+    console.log('Stack not found');
+    return;
+  }
+
+  await updateBucketPolicy('App', details.appBucket, details.appDistribution, details.appOriginAccessIdentity, options);
+
+  await updateBucketPolicy(
+    'Storage',
+    details.storageBucket,
+    details.storageDistribution,
+    details.storageOriginAccessIdentity,
+    options
+  );
+
+  console.log('Done');
+}
+
+export async function updateBucketPolicy(
+  friendlyName: string,
+  bucketResource: StackResource | undefined,
+  distributionResource: StackResource | undefined,
+  oaiResource: StackResource | undefined,
+  options: UpdateBucketPoliciesOptions
+): Promise<void> {
+  if (!bucketResource?.PhysicalResourceId) {
+    console.log(`${friendlyName} bucket not found`);
+    return;
+  }
+
+  if (!distributionResource?.PhysicalResourceId) {
+    console.log(`${friendlyName} distribution not found`);
+    return;
+  }
+
+  if (!oaiResource?.PhysicalResourceId) {
+    console.log(`${friendlyName} OAI not found`);
+    return;
+  }
+
+  const bucketName = bucketResource.PhysicalResourceId;
+  const oaiId = oaiResource.PhysicalResourceId;
+  const bucketPolicy = await getPolicy(bucketName);
+  if (policyHasStatement(bucketPolicy, bucketName, oaiId)) {
+    console.log(`${friendlyName} bucket already has policy statement`);
+    return;
+  }
+
+  addPolicyStatement(bucketPolicy, bucketName, oaiId);
+  console.log(`${friendlyName} bucket policy:`);
+  console.log(JSON.stringify(bucketPolicy, undefined, 2));
+
+  if (options.dryrun) {
+    console.log('Dry run - skipping updates');
+  } else {
+    // Apply the updated policy
+    console.log('Updating bucket policy...');
+    await setPolicy(bucketName, bucketPolicy);
+    console.log('Bucket policy updated');
+
+    // Create a CloudFront invalidation to clear any cached responses
+    console.log('Creating CloudFront invalidation...');
+    await createInvalidation(distributionResource.PhysicalResourceId);
+    console.log('CloudFront invalidation created');
+  }
+
+  console.log(`${friendlyName} bucket policy updated`);
```

Comment:
This might be confusing for users in dry-run mode, consider moving this log into the `else` case